src/genie/libs/parser/junos/show_platform.py

```diff
@@ -113,3 +114,70 @@ def cli(self, filename='', output=None):
 
         return ret_dict
 
+# ===========================
+# Parser for show version
+# ===========================
+
+class ShowVersionSchema(MetaParser):
+    """Schema for show version"""
+    schema = {'hostname': str,
+              'operating_system': str,
+              'software_version': str,
+              'model': str,
+              }
+
+
+# =========================
+# Parser for 'show version'
+# =========================
+
+
+class ShowVersion(ShowVersionSchema):
+    """Parser for show version"""
+    cli_command = 'show version'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        # Init vars
+        show_version_dict = {}
+
+        # regex patterns
+
+        # Junos: 15.1R1-S1
+        p1 = re.compile('\s*Junos: +(?P<software_version>.*$)')
```

Comment:
after `line.strip()`, there is no space at the beginning and the end of the line
so the regex usually looks like `re.compile(r'^Junos: +(?P<junos_version>\S+)$')`

please update other regexes as well